EIPS/eip-5593.md

```diff
@@ -0,0 +1,84 @@
+---
+eip: 5593
+title: Restrict Web3 Provider Object API Injection
+description: Wallet guidance for restricting web3 provider object API access to secure contexts for improved privacy and security for wallet users.
+author: Yan Zhu (@diracdeltas), Brian R. Bondy (@bbondy), Kyle Den Hartog (@kdenhartog)
+discussions-to: https://ethereum-magicians.org/t/rfc-limiting-provider-object-injection-to-secure-contexts/10670
+status: Draft
+type: Standards Track
+category: Interface
+created: 2022-09-05
+requires: 1193
+---
+
+## Abstract
+
+Historically the web platform has had a notion of “powerful” APIs like [geolocation](https://www.w3.org/TR/geolocation/) and [camera/microphone](https://www.w3.org/TR/mediacapture-streams/), which are subject to additional security restrictions such as those defined by [secure contexts](https://www.w3.org/TR/secure-contexts/). Since the web3 provider object APIs allow dApp websites to request access to sensitive user data and to request use of user funds, new web3 provider object APIs generally should align to the security considerations of other powerful APIs like geolocation.
+
+## Motivation
+
+Wallets are oftentimes maintaining security and safety of users' funds that can be equivalent to large portions of money. For this reason, it's a good idea to restrict access to the APIs that grant this access in certain contexts to align it more closely with the principles followed for web platform APIs given web3 provider object APIs are often extending this platform via extensions. By adding in restrictions to where extensions inject their APIs, we're reducing the surface area that malicious web pages could fingerprint or falsely request access to the web3 provider object APIs to harm users by accessing their data or stealing their web3 assets.
+
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+### Restrictions for providers
+
+The provider objects, e.g. `window.ethereum`, are not provided in all contexts. The following restrictions are REQUIRED for conformant wallets:
```

Comment:
> The provider objects, e.g. `window.ethereum`, are not provided in all contexts.

Are you saying they currently aren't provided in all contexts, or that they shouldn't be provided in all contexts by wallets following this standard?

Reply:
> they shouldn't be provided in all contexts by wallets following this standard?

Meaning this one. Good catch this language could use to be updated

Reply:
```suggestion
In some contexts, Ethereum provider objects (such as `window.ethereum`) MUST NOT be provided. Wallets MUST obey the following restrictions:
```
Nit: EIPs are always optional to implement, and it is up to the various implementations to choose to apply or not apply any given EIP. The usage of the word "compliant" is therefore redundant when it refers to the current EIP.

Reply:
Two points:

1. The ambiguity of the MUST NOT normative language update here is an untestable statement. 

2. The purpose of this statement is to say that ALL of the following statements MUST be implemented for this to be conformant. It's not conforming with this spec to pick and choose the rules you like and don't like here as that would lead to compatibility issues between wallets.